hydra-node/test/Hydra/JSONSchema.hs

```diff
@@ -132,22 +182,26 @@ type SpecificationSelector = Traversal' Aeson.Value Aeson.Value
 -- But tools (and in particular jsonschema) only works from JSON, so this
 -- function makes sure to also convert our local yaml into JSON.
 withJsonSpecifications ::
-  (FilePath -> IO ()) ->
-  IO ()
+  MonadIO m =>
+  (FilePath -> m r) ->
+  m r
 withJsonSpecifications action = do
-  specDir <- (</> "json-schemas") . normalise <$> Pkg.getDataDir
-  specFiles <- listDirectory specDir
-  dir <- createSystemTempDirectory "Hydra_APISpec"
-  forM_ specFiles $ \file -> do
-    when (takeExtension file == ".yaml") $ do
-      spec <- Yaml.decodeFileThrow @_ @Aeson.Value (specDir </> file)
-      let spec' = addField "$id" ("file://" <> dir <> "/") spec
-      Aeson.encodeFile (dir </> takeBaseName file <.> "yaml") spec'
-  action dir
- where
-  addField :: ToJSON a => Aeson.Key -> a -> Aeson.Value -> Aeson.Value
-  addField k v = withObject (at k ?~ toJSON v)
+  specDir <- (</> "json-schemas") . normalise <$> liftIO Pkg.getDataDir
+  specFiles <- liftIO $ listDirectory specDir
+  withTempDir "Hydra_APISpec" $ \dir -> do
+    forM_ specFiles $ \file -> do
+      when (takeExtension file == ".yaml") $ do
+        spec <- Yaml.decodeFileThrow @_ @Aeson.Value (specDir </> file)
+        let spec' = addField "$id" ("file://" <> dir <> "/") spec
+        liftIO $ Aeson.encodeFile (dir </> takeBaseName file <.> "json") spec'
+        -- XXX: We need to write the specFile as .yaml as well because internally the
+        -- spec .json reference elements within the yaml file.
```

Comment:
```suggestion
        -- XXX: We need to write the specFile as .yaml although it is a JSON document now, 
        -- because internally the spec reference elements using the original .yaml file name.
```